snuba/querylog/query_metadata.py

```diff
@@ -15,6 +22,107 @@ class QueryStatus(Enum):
     TIMEOUT = "timeout"
 
 
+QUERY_ERROR_MAPPINGS = {
+    ErrorCodes.TOO_SLOW: QueryStatus.TIMEOUT,
+    ErrorCodes.TIMEOUT_EXCEEDED: QueryStatus.TIMEOUT,
+    ErrorCodes.SOCKET_TIMEOUT: QueryStatus.TIMEOUT,
+    ErrorCodes.NETWORK_ERROR: QueryStatus.TIMEOUT,
+}
+
+
+def get_query_status_from_error_codes(code: ErrorCodes) -> QueryStatus | None:
+    return QUERY_ERROR_MAPPINGS.get(code)
+
+
+class SLOStatus(Enum):
+    """
+    The different statuses we return for a request.
+
+    TODO: This will replace QueryStatus, but both exist as we cut over.
+    """
+
+    # Successfully returned a response
+    SUCCESS = "success"
+    # Failed validation
+    INVALID_REQUEST = "invalid-request"
+    # A type mismatch between arguments, ILLEGAL_TYPE_OF_ARGUMENT/TYPE_MISMATCH
+    INVALID_TYPING = "invalid-typing"
+    # Rejected because of one of the product based rate limiters
+    RATE_LIMITED = "rate-limited"
+    # Rejected by a table rate limiter
+    TABLE_RATE_LIMITED = "table-rate-limited"
+    # If the thread responsible for executing a query and writing it to the cache times out
+    CACHE_SET_TIMEOUT = "cache-set-timeout"
+    # The thread waiting for a cache result to be written times out
+    CACHE_WAIT_TIMEOUT = "cache-wait-timeout"
+    # Clickhouse predicted the query would time out. TOO_SLOW
+    PREDICTED_TIMEOUT = "predicted-timeout"
+    # Clickhouse timeout, TIMEOUT_EXCEEDED
+    CLICKHOUSE_TIMEOUT = "clickhouse-timeout"
+    # Network isssues, SOCKET_TIMEOUT NETWORK_ERROR
+    NETWORK_TIMEOUT = "network-timeout"
+    # Query used too much memory, MEMORY_LIMIT_EXCEEDED
+    MEMORY_EXCEEDED = "memory-exceeded"
+    # Any other error
+    ERROR = "error"
+
+
+class SLO(Enum):
```

Comment:
I don't love these names but I don't know what to call them. Open to suggestions.